test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationParts/ApplicationPartManagerTest.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Core.Test.ApplicationParts
+{
+    public class ApplicationPartManagerTest
+    {
+
+        [Fact]
+        public void PopulateFeature_InvokesAllProvidersSequentially_ForAGivenFeature()
+        {
+            // Arrange
+            var manager = new ApplicationPartManager();
+            manager.ApplicationParts.Add(new TestPart());
+            manager.ApplicationParts.Add(new OtherPartType());
+            manager.ApplicationParts.Add(new TestPart());
+            manager.Providers.Add(new FeatureProvider(f => f.Value++));
+            manager.Providers.Add(new FeatureProvider(f => f.Value = f.Value * 3));
+
+            var feature = new Feature();
+            // Act
+            manager.PopulateFeature(feature);
+
+            // Assert
+            Assert.Equal(18, feature.Value);
```

Comment:
This seems like some fun math, but hard to follow. Can this use some plain old assertions or something? Or record all the calls and compare against to a known list of calls?